### DISCUSSION TITLE
List all security roles in an environment

### DISCUSSION THREAD

**Discussion by Laskewitz** [microsoft/powerplatform-build-tools#474] [Ideas]
In the `pac admin assign-group` and `pac admin assign-user` commands, there is a parameter called `--role` which you can use to add a security role name or ID that will assign the role to that user or group. 

# Example command(s)

It would be nice to have a command to list all security roles in an environment:

`pac admin list-roles --json` - list the security roles in the current environment and output the result in JSON format.
`pac admin list-roles --environment [guid]` - list the security roles in the provided environment.
`pac admin list-roles --filter [string]` - list the security roles in the current environment filtered by the string filter.

This could be really useful.

**Comment by anttipajunen** [comment]
I'd add that listing the sec roles per BU would be important so that we can distinguish roles between BUs. The output should definitely be in JSON.

**Comment by resc-ambit** [comment]
The absence of such a command encourage (lazy) users to leave the --role parameter (which is optional) and results in assigning the System Administrator role.

Temporary workaround:
`pac org fetch -x "<fetch><entity name="role" ><filter><condition attribute="name" operator="like" value="%myrolename%" /></filter><attribute name="name" /><attribute name="roleid" /><attribute name="businessunitid" /></entity></fetch>"`

**Comment by parvezghumra** [comment]
Looks like this available in v2.5.1 now :)

https://www.nuget.org/packages/Microsoft.PowerApps.CLI#releasenotes-body-tab

**Reply by Laskewitz** [reply]
Yup - spotted it as well! Good news!
